usagov/usagov-2021
# Usagov 352 synonyms

Created a new entity type agency synonym which is shown in agency view. Synonyms display and link to parent information. 


https://cm-jira.usa.gov/browse/USAGOV-352

## Description


Created a new entity type: Agency Synonym:
- - Agency Synonym has one field: Agency Reference. 
- - - Agency Reference is Required Field Type: Entity Reference -> Content limited to 1
- - - Machine Name:  field_agency_reference	
- - - Reference type -> Reference method: Default 
- - - Reference type -> content type: Federal Directory Record

Changed Federal Agencies View: 
- Added filter criteria content type = agency synonym
- Added new filter group & Put new filter in it’s own group separate groups with OR

Created new file usagov_directories.module:
- Pre render hook for agency view
- - For every result that is a synonym; shallow copy it's parent agency and display that information instead

Testing:
1. Rebuild of theme registry is required 
2. Flush all cache 
3. Create a couple new agencies
4. Create a few synonyms for said agencies (I like to also test synonyms that don't start with the same letter as parent agency)
5. See if synonym is displayed in view
6. See if synonym correctly displays parent information  

Sample Testing Inputs:

- Agency 
- - Synonym 1
- - Synonym 2 

- Agency Placeholder
- - AP
- Food and Drug Administration
- - FDA
- - Foods
- - Eats and Meds
- Census Bureau
- - Bureau of the Census
- - Census
- - U.S. Census Bureau 



## Checklist for the Developer



## Checklist for the Peer Reviewers
